src/main/scala/scorex/core/network/message/Message.scala

```diff
@@ -79,7 +87,11 @@ class MessageSerializer(specs: Seq[MessageSpec[_]], magicBytes: Array[Byte]) {
           val checksum = it.getBytes(ChecksumLength)
           val data = it.getBytes(length)
           val digest = Blake2b256.hash(data).take(ChecksumLength)
-          if (!java.util.Arrays.equals(checksum, digest)) throw new Error(s"Invalid data checksum length = $length")
+
+          //peer reported incorrect checksum
+          if (!java.util.Arrays.equals(checksum, digest)) {
+            throw MaliciousBehaviorException(s"Invalid data checksum length = $length")
```

Comment:
The error message seems off. Why `length` is being reported when `checksum != digest`?

Reply:
Thanks, fixed!